pkg/cpio/types.go

```diff
@@ -33,27 +34,61 @@ type RecordFormat interface {
 	Writer(w io.Writer) RecordWriter
 }
 
-func StaticRecord(contents string, info Info) Record {
+func StaticRecord(contents []byte, info Info) Record {
 	info.FileSize = uint64(len(contents))
 	return Record{
-		bytes.NewReader([]byte(contents)),
-		info,
+		ReadCloser: NewBytesReadCloser(contents),
+		Info:       info,
 	}
 }
 
-type eofReader struct{}
+type ReadCloser struct {
+	io.Reader
+}
 
-func (eofReader) Read(p []byte) (int, error) {
-	return 0, io.EOF
+func (b ReadCloser) Close() error {
+	return nil
+}
+
+func NewBytesReadCloser(contents []byte) io.ReadCloser {
+	return &ReadCloser{bytes.NewReader(contents)}
+}
+
+func NewReadCloser(r io.Reader) io.ReadCloser {
+	return &ReadCloser{r}
 }
 
 func EmptyRecord(info Info) Record {
 	return Record{
-		eofReader{},
+		NewBytesReadCloser([]byte{}),
 		info,
 	}
 }
 
+type DeferReader struct {
```

Comment:
DeferReader makes me think of the Go keyword `defer` and that doesn't entirely feel accurate here.

Can we call it LazyReader?